### PR TITLE
SYS-3521 SYS-3522 Open source avn runtime - part 1

### PR DESCRIPTION
**Pull request by nahuseyoum**
This PR open sources the runtime code only. CI and build are not expected to work but will be fixed with subsequent PRs.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
+authors = ["Aventus systems team"]
+homepage = "https://www.aventus.io/"
+repository = "https://github.com/Aventus-Network-Services/avn-node-parachain/"
+edition = "2021"
+rust-version = "1.64.0"
+
 [workspace]
 members = [
     "pallets/*",
-    "primitives/*",
```
Comment (thadouk) [inline Cargo.toml:7]: why remove primitives?

File: runtime/test/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,1016 @@
...
+	type WeightInfo = pallet_parachain_staking::weights::SubstrateWeight<Runtime>;
+}
+
+// Substrate pallets that AvN has dependency
+impl pallet_authority_discovery::Config for Runtime {
+	type MaxAuthorities = ConstU32<100_000>;
+}
+
+impl pallet_session::historical::Config for Runtime {
+	// TODO review this as originally was using the staking pallet. This is a minimal approach on the Identification
```
Comment (thadouk) [inline runtime/test/src/lib.rs:493]: same as avn runtime

File: runtime/test/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,1016 @@
...
+impl pallet_avn::Config for Runtime {
+	type AuthorityId = AvnId;
+	type EthereumPublicKeyChecker = ValidatorsManager;
+	type NewSessionHandler = ValidatorsManager;
+	type DisabledValidatorChecker = ValidatorsManager;
+	type FinalisedBlockChecker = AvnFinalityTracker;
+}
+
+parameter_types! {
+	// TODO [TYPE: review][PRI: medium][JIRA: SYS-358]: Configurable in eth-events pallet?
```
Comment (thadouk) [inline runtime/test/src/lib.rs:560]: same as avn-runtime

**Review by thadouk — APPROVED** [review]
Approved provisionally.
Primitives should be restored on the cargo.toml and some code comments should be removed before publishing.

File: runtime/avn/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,1006 @@
...
+	type WeightInfo = pallet_parachain_staking::weights::SubstrateWeight<Runtime>;
+}
+
+// Substrate pallets that AvN has dependency
+impl pallet_authority_discovery::Config for Runtime {
+	type MaxAuthorities = ConstU32<100_000>;
+}
+
+impl pallet_session::historical::Config for Runtime {
+	// TODO review this as originally was using the staking pallet. This is a minimal approach on the Identification
```
[the remark this answers is not in the document]
Comment (nahuseyoum) [reply runtime/avn/src/lib.rs:475]: Yes I think it's still relevant.

File: runtime/avn/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,1006 @@
...
+impl pallet_avn::Config for Runtime {
+	type AuthorityId = AvnId;
+	type EthereumPublicKeyChecker = ValidatorsManager;
+	type NewSessionHandler = ValidatorsManager;
+	type DisabledValidatorChecker = ValidatorsManager;
+	type FinalisedBlockChecker = AvnFinalityTracker;
+}
+
+parameter_types! {
+	// TODO [TYPE: review][PRI: medium][JIRA: SYS-358]: Configurable in eth-events pallet?
```
[the remark this answers is not in the document]
Comment (nahuseyoum) [reply runtime/avn/src/lib.rs:542]: I also think we should keep this. Making it configurable in code would be more flexible.

File: runtime/avn/src/weights/block_weights.rs [hunk 6]
```diff
@@ -0,0 +1,53 @@
+// This file is part of Substrate.
```
[the remark this answers is not in the document]
Comment (nahuseyoum) [reply runtime/avn/src/weights/block_weights.rs:1]: We can but that's a separate change.

File: Cargo.toml (hunk 1 above)
Comment (nahuseyoum) [reply Cargo.toml:7]: We shouldn't, this is a merge conflict. Will restore it

File: runtime/avn/src/weights/block_weights.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (thadouk) [reply runtime/avn/src/weights/block_weights.rs:1]: I think it is simple to do and would reduce the PR size, reduce the codebase and avoid code repetition.
Falls well under the preparing the runtime code for open sourcing task.

[the remark this answers is not in the document]
Comment (nahuseyoum) [reply runtime/avn/src/weights/block_weights.rs:1]: I agree with the reason to do it, will add a separate ticket for it but It doesn't fall under the `preparing the runtime code for open sourcing` task.

[the remark this answers is not in the document]
Comment (thadouk) [reply runtime/avn/src/weights/block_weights.rs:1]: I don't see the need for a new task, it is well within the scope of this task where we review and cleanup up the codebase.

**Pull request merged**